contracts/src/cryptography/message_hash_utils.rs

```diff
@@ -0,0 +1,93 @@
+//! Signature message hash utilities for producing digests to be consumed by
+//! `ECDSA` recovery or signing.
+//!
+//! The library provides methods for generating a hash of a message that
+//! conforms to the [ERC-191] and [EIP 712] specifications.
+//!
+//! [ERC-191]: https://eips.ethereum.org/EIPS/eip-191
+//! [EIP-712]: https://eips.ethereum.org/EIPS/eip-712
+
+use alloc::string::String;
+
+use alloy_primitives::{bytes, keccak256, B256};
+use alloy_sol_types::{sol, SolType};
+
+type EthMessageCoder = sol! {
+    tuple(bytes, bytes32)
+};
+
+type TypeHashCoder = sol! {
+    tuple(string, bytes32, bytes32)
+};
+
+/// Returns the keccak256 digest of an ERC-191 signed data with version `0x45`
+/// (`personal_sign` messages).
+///
+/// The digest is calculated by prefixing a bytes32 `message_hash` with
+/// `"\x19Ethereum Signed Message:\n32"` and hashing the result. It corresponds
+/// with the hash signed when using the [eth_sign] JSON-RPC method.
+///
+/// NOTE: The `message_hash` parameter is intended to be the result of hashing a
+/// raw message with keccak256, although any bytes32 value can be safely used
+/// because the final digest will be re-hashed.
+///
+/// [eth_sign]: https://eth.wiki/json-rpc/API#eth_sign
+pub fn to_eth_signed_message_hash(message_hash: B256) -> B256 {
```

Comment:
This is not doing exactly what the comment says. Let's try to replicate [this code instead](https://github.com/alloy-rs/core/blob/adcf7adfa1f35c56e6331bab85b8c56d32a465f1/crates/primitives/src/utils/mod.rs#L111-L143).

Reply:
Note that we append `32` because that is the message length, but we should have the more general `eip191_message` and then use it as it should be used in this particular case for messages of length 32.

Reply:
sorry the comment was wrong.

```solidity
    function toEthSignedMessageHash(bytes32 messageHash) internal pure returns (bytes32 digest) {
        /// @solidity memory-safe-assembly
        assembly {
            mstore(0x00, "\x19Ethereum Signed Message:\n32") // 32 is the bytes-length of messageHash
            mstore(0x1c, messageHash) // 0x1c (28) is the length of the prefix
            digest := keccak256(0x00, 0x3c) // 0x3c is the length of the prefix (0x1c) + messageHash (0x20)
        }
    }
```

this is the solidity code I am trying to emulate;

Reply:
looking at this; 

https://github.com/developeruche/rust-contracts-stylus/blob/8d0954f9eb7de2a4056794b19029e725dcf61a8d/lib/crypto/src/message_hash_utils.rs#L31

should I still proceed?

Reply:
I think that looks good 👍

Reply:
alright